lib/compute/index.js

```diff
@@ -0,0 +1,600 @@
+ * @param {number=} options.maxResults - Maximum number of instances to 
+ *     return.
+ * @param {string=} options.zone - Only instances in this zone a returned.
+ * @param {object=} options.filter - Search filter.
+ * @param {string} options.filter.fieldName - Instance field to consider in this
+ *     filter.
+ * @param {string} options.filter.operatorString - Filter operator, can be
+ *     either 'eq' or 'ne'.
+ * @param {string} options.filter.literalString - String to compare the instance
+ *     field to. Can be a regular expression.
+ * @param {string=} pageToken - Page identifier used in paginated search.
+ * @param {function} callback - The callback function.
+ *
+ * @example
+ * var callback = function(err, instances) {
+ *   // `instances` is an array of `Instance` objects.
+ * };
+ *
+ * var compute = gcloud.compute({
+ *   projectId: 'grape-spaceship-123'
+ * });
+ *
+ * compute.listInstances( 
+ *   {
+ *     zone: 'europe-west1-b',
+ *     filter: {
+ *       fieldName : 'name',
+ *       operatorString : 'eq',
+ *       literalString : 'instance-[0-9]'
+ *     }],
+ *   }, callback);
+ */
+Compute.prototype.listInstances = function(options, callback) {
+  if (!callback) {
+    callback = options;
+  }
+
+  var query = {};
+
+  if (options) {
+    if (util.is(options.maxResults, 'number')) {
+      query.maxResults = options.maxResults;
+    }
+    if (options.filter) {
+      if (!util.is(options.filter.fieldName, 'string') ||
+          !util.is(options.filter.operatorString, 'string') ||
+          !util.is(options.filter.literalString, 'string')) {
+        throw new Error(
+          'A filter must have name, comparisonString and literal fields.');
+      }
+      if (options.filter.operatorString !== 'eq' && 
+        options.filter.operatorString !== 'ne') {
+        throw new Error('Filter operator must be either eq or ne');
+      }
+      query.filter = 
+        options.filter.fieldName + ' ' +
+        options.filter.operatorString + ' \'' +
+        options.filter.literalString + '\'';
+    }
+    if (util.is(options.pageToken, 'string')) {
+      query.pageToken = options.pageToken;
+    }
+  }
+  
+  this.makeReq_(
+    'GET', 
+    '/aggregated/instances', 
+    query, 
+    null, function(err, resp) {
+      if (err) {
+        callback(err);
+        return; 
+      }
+      if (resp.items) {
+        var instances = [];
+        for (var zoneName in resp.items) {
+          var zone = this.zone(zoneName);
+          var zoneInstances = resp.items[zoneName].instances;
+          if (zoneInstances) {
+            for (var i = 0; i < zoneInstances.length; i++) {
+              var instance = new Instance(
+                zone, 
+                zoneInstances[i].name, 
+                zoneInstances[i]);
+              instances.push(instance);
+            }
+          }
+        }
+        callback(null, instances);    
+      } else {
+        callback(null, []);
+      }
+    }.bind(this));
```

Comment:
While I do love `.bind()` I think our style guide discourages the use of it.

https://github.com/GoogleCloudPlatform/gcloud-node/issues/727#issuecomment-122353264

If you rebase your branch we added a style linter to the `npm run lint` command